0.27.0 crashing after startup

### Current Behavior

After updating to 0.27.0 windows_exporter service keeps crashing.

### Expected Behavior

Should run just fine

### Steps To Reproduce

_No response_

### Environment

- windows_exporter Version: 0.27.0
- Windows Server Version: 2019


### windows_exporter logs

```shell
Der Dienst "windows_exporter" wurde unerwartet beendet. Dies ist bereits 1 Mal vorgekommen. Folgende Korrekturmaßnahmen werden in 60000 Millisekunden durchgeführt: Neustart des Diensts.
```


### Anything else?

_No response_

I set the --log.file to stderr and got the following:
```
PS C:\Program Files\windows_exporter> .\windows_exporter.exe --log.file stderr --config.file="c:\monitoring\windows_exporter\windows_exporter.yml" --config.file.insecure-skip-verify --log.level=debug
ts=2024-08-13T17:37:53.350Z caller=exporter.go:147 level=debug msg="Logging has Started"
ts=2024-08-13T17:37:53.351Z caller=config.go:75 level=info msg="Loading configuration file: c:\\monitoring\\windows_exporter\\windows_exporter.yml"
ts=2024-08-13T17:37:53.355Z caller=wmi.go:16 level=debug msg="Initializing SWbemServices"
ts=2024-08-13T17:37:53.393Z caller=iis.go:925 level=debug collector=iis msg="Detected IIS 8.5\n"
ts=2024-08-13T17:37:53.396Z caller=service.go:102 level=warn collector=service msg="No where-clause specified for service collector. This will generate a very large number of metrics!"
ts=2024-08-13T17:37:53.399Z caller=mssql.go:81 level=debug collector=mssql msg="Detected MSSQL Instances: mssql.mssqlInstancesType{\"MSSQLSERVER\":\"MSSQL14.MSSQLSERVER\"}\n"
ts=2024-08-13T17:37:53.671Z caller=exporter.go:222 level=info msg="Running as #########"
ts=2024-08-13T17:37:53.672Z caller=exporter.go:229 level=info msg="Enabled collectors: memory, process, cpu, cpu_info, physical_disk, cs, mssql, os, system, diskdrive, logical_disk, iis, net, service"
ts=2024-08-13T17:37:53.674Z caller=exporter.go:264 level=info msg="Starting windows_exporter" version="(version=0.27.0, branch=HEAD, revision=ca4ad46e2df498e0317d09bc2c037922cd879898)"
ts=2024-08-13T17:37:53.676Z caller=exporter.go:265 level=info msg="Build context" build_context="(go=go1.22.6, platform=windows/amd64, user=runneradmin@fv-az1495-484, date=20240811-13:54:18, tags=unknown)"
ts=2024-08-13T17:37:53.678Z caller=exporter.go:266 level=debug msg="Go MAXPROCS" procs=16
ts=2024-08-13T17:37:53.681Z caller=tls_config.go:313 level=info msg="Listening on" address=[::]:9182
ts=2024-08-13T17:37:53.682Z caller=tls_config.go:316 level=info msg="TLS is disabled." http2=false address=[::]:9182
ts=2024-08-13T17:38:18.715Z caller=prometheus.go:189 level=debug msg="collector memory succeeded after 0.000000s."
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x97fa2b]

goroutine 83 [running]:
github.com/prometheus-community/windows_exporter/pkg/collector/mssql.(*Collector).execute(0xc000249008, 0xc00038e0b0, {0xc000048e00, 0x75}, 0x0, 0xc00004a480, {0xc00039bbb0, 0xb}, 0x0?)
        D:/a/windows_exporter/windows_exporter/pkg/collector/mssql/mssql.go:1979 +0xcb
created by github.com/prometheus-community/windows_exporter/pkg/collector/mssql.(*Collector).Collect in goroutine 67
        D:/a/windows_exporter/windows_exporter/pkg/collector/mssql/mssql.go:2020 +0x1dc
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x97fa2b]

goroutine 82 [running]:
github.com/prometheus-community/windows_exporter/pkg/collector/mssql.(*Collector).execute(0xc000249008, 0xc00038e0b0, {0x0, 0x0}, 0x0, 0xc00004a480, {0xc00039bbb0, 0xb}, 0xc000098ff0?)
        D:/a/windows_exporter/windows_exporter/pkg/collector/mssql/mssql.go:1979 +0xcb
created by github.com/prometheus-community/windows_exporter/pkg/collector/mssql.(*Collector).Collect in goroutine 67
        D:/a/windows_exporter/windows_exporter/pkg/collector/mssql/mssql.go:2020 +0x1dc
PS C:\Program Files\windows_exporter>
```

Here is my config file:
```
---
collectors:
  enabled: cpu,cpu_info,cs,diskdrive,logical_disk,iis,memory,mssql,net,os,physical_disk,process,service,system
  mssql:
    classes-enabled:
      - accessmethods
      - availreplica
      - bufman
      - databases
      - dbreplica
      - genstats
      - locks
      - memmgr
      - sqlstats
      - sqlerrors
      - transactions
      - waitstats
scrape:
  timeout-margin: 0.5
telemetry:
  path: /metrics
  max-requests: 5
web:
  listen-address: ":9182"
process:
  priority: normal
```

--- a/pkg/collector/dfsr/dfsr.go
+++ b/pkg/collector/dfsr/dfsr.go
@@ -120,10 +120,17 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	c := &Collector{
 		config: ConfigDefaults,
 	}
-	c.config.CollectorsEnabled = make([]string, 0)
+
+	var collectorsEnabled string
 
 	app.Flag("collectors.dfsr.sources-enabled", "Comma-separated list of DFSR Perflib sources to use.").
-		Default(strings.Join(ConfigDefaults.CollectorsEnabled, ",")).StringsVar(&c.config.CollectorsEnabled)
+		Default(strings.Join(ConfigDefaults.CollectorsEnabled, ",")).StringVar(&collectorsEnabled)
+
+	app.Action(func(*kingpin.ParseContext) error {
+		c.config.CollectorsEnabled = strings.Split(collectorsEnabled, ",")
+
+		return nil
+	})
 
 	return c
 }

--- a/pkg/collector/exchange/exchange.go
+++ b/pkg/collector/exchange/exchange.go
@@ -111,6 +111,7 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	c.config.CollectorsEnabled = make([]string, 0)
 
 	var listAllCollectors bool
+	var collectorsEnabled string
 
 	app.Flag(
 		"collectors.exchange.list",
@@ -120,7 +121,7 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	app.Flag(
 		"collectors.exchange.enabled",
 		"Comma-separated list of collectors to use. Defaults to all, if not specified.",
-	).Default(strings.Join(ConfigDefaults.CollectorsEnabled, ",")).StringsVar(&c.config.CollectorsEnabled)
+	).Default(strings.Join(ConfigDefaults.CollectorsEnabled, ",")).StringVar(&collectorsEnabled)
 
 	app.PreAction(func(*kingpin.ParseContext) error {
 		if listAllCollectors {
@@ -148,6 +149,12 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 
 			os.Exit(0)
 		}
+
+		return nil
+	})
+
+	app.Action(func(*kingpin.ParseContext) error {
+		c.config.CollectorsEnabled = strings.Split(collectorsEnabled, ",")
 
 		return nil
 	})

--- a/pkg/collector/mssql/mssql.go
+++ b/pkg/collector/mssql/mssql.go
@@ -6,7 +6,6 @@ import (
 	"errors"
 	"fmt"
 	"os"
-	"slices"
 	"sort"
 	"strings"
 	"sync"
@@ -437,9 +436,9 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	c := &Collector{
 		config: ConfigDefaults,
 	}
-	c.config.CollectorsEnabled = make([]string, 0)
 
 	var listAllCollectors bool
+	var collectorsEnabled string
 
 	app.Flag(
 		"collectors.mssql.class-print",
@@ -449,7 +448,7 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	app.Flag(
 		"collectors.mssql.classes-enabled",
 		"Comma-separated list of mssql WMI classes to use.",
-	).Default(strings.Join(ConfigDefaults.CollectorsEnabled, ",")).StringsVar(&c.config.CollectorsEnabled)
+	).Default(strings.Join(c.config.CollectorsEnabled, ",")).StringVar(&collectorsEnabled)
 
 	app.PreAction(func(*kingpin.ParseContext) error {
 		if listAllCollectors {
@@ -468,6 +467,12 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 		return nil
 	})
 
+	app.Action(func(*kingpin.ParseContext) error {
+		c.config.CollectorsEnabled = strings.Split(collectorsEnabled, ",")
+
+		return nil
+	})
+
 	return c
 }
 
@@ -480,16 +485,11 @@ func (c *Collector) SetLogger(logger log.Logger) {
 }
 
 func (c *Collector) GetPerfCounter() ([]string, error) {
-	enabled := slices.Compact(c.config.CollectorsEnabled)
-
-	// Result must order, to prevent test failures.
-	sort.Strings(enabled)
-
 	c.mssqlInstances = getMSSQLInstances(c.logger)
-	perfCounters := make([]string, 0, len(c.mssqlInstances)*len(enabled))
+	perfCounters := make([]string, 0, len(c.mssqlInstances)*len(c.config.CollectorsEnabled))
 
 	for instance := range c.mssqlInstances {
-		for _, c := range enabled {
+		for _, c := range c.config.CollectorsEnabled {
 			perfCounters = append(perfCounters, mssqlGetPerfObjectName(instance, c))
 		}
 	}
@@ -502,6 +502,9 @@ func (c *Collector) Close() error {
 }
 
 func (c *Collector) Build() error {
+	// Result must order, to prevent test failures.
+	sort.Strings(c.config.CollectorsEnabled)
+
 	// meta
 	c.mssqlScrapeDurationDesc = prometheus.NewDesc(
 		prometheus.BuildFQName(types.Namespace, Name, "collector_duration_seconds"),
@@ -1965,6 +1968,12 @@ func (c *Collector) Build() error {
 
 	c.mssqlCollectors = c.getMSSQLCollectors()
 
+	for _, name := range c.config.CollectorsEnabled {
+		if _, ok := c.mssqlCollectors[name]; !ok {
+			return errors.New("unknown mssql collector: " + name)
+		}
+	}
+
 	return nil
 }
 
@@ -2007,13 +2016,8 @@ func (c *Collector) execute(ctx *types.ScrapeContext, name string, fn mssqlColle
 func (c *Collector) Collect(ctx *types.ScrapeContext, ch chan<- prometheus.Metric) error {
 	wg := sync.WaitGroup{}
 
-	enabled := slices.Compact(c.config.CollectorsEnabled)
-
-	// Result must order, to prevent test failures.
-	sort.Strings(enabled)
-
 	for sqlInstance := range c.mssqlInstances {
-		for _, name := range enabled {
+		for _, name := range c.config.CollectorsEnabled {
 			function := c.mssqlCollectors[name]
 
 			wg.Add(1)

--- a/pkg/collector/textfile/textfile.go
+++ b/pkg/collector/textfile/textfile.go
@@ -78,12 +78,19 @@ func NewWithFlags(app *kingpin.Application) *Collector {
 	c := &Collector{
 		config: ConfigDefaults,
 	}
-	c.config.TextFileDirectories = make([]string, 0)
+
+	var textFileDirectories string
 
 	app.Flag(
 		"collector.textfile.directories",
 		"Directory or Directories to read text files with metrics from.",
-	).Default(strings.Join(ConfigDefaults.TextFileDirectories, ",")).StringsVar(&c.config.TextFileDirectories)
+	).Default(strings.Join(ConfigDefaults.TextFileDirectories, ",")).StringVar(&textFileDirectories)
+
+	app.Action(func(*kingpin.ParseContext) error {
+		c.config.TextFileDirectories = strings.Split(textFileDirectories, ",")
+
+		return nil
+	})
 
 	return c
 }

